spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/subscriber/PubSubSubscriberTemplate.java

```diff
@@ -164,12 +172,62 @@ private void ack(String subscriptionName, Collection<String> ackIds) {
 	}
 
 	private void nack(String subscriptionName, Collection<String> ackIds) {
+		modifyAckDeadline(subscriptionName, ackIds, 0);
+	}
+
+	private void modifyAckDeadline(String subscriptionName, Collection<String> ackIds, int ackDeadlineSeconds) {
 		ModifyAckDeadlineRequest modifyAckDeadlineRequest = ModifyAckDeadlineRequest.newBuilder()
-				.setAckDeadlineSeconds(0)
+				.setAckDeadlineSeconds(ackDeadlineSeconds)
 				.addAllAckIds(ackIds)
 				.setSubscription(subscriptionName)
 				.build();
 
 		this.subscriberStub.modifyAckDeadlineCallable().call(modifyAckDeadlineRequest);
 	}
+
+
+	private class PulledAcknowledgeablePubsubMessage implements AcknowledgeablePubsubMessage {
```

Comment:
I would like to see a `toString()` on this class - pretty useful during application debugging

Reply:
I see that this isn't a `private static class` because of the `PubSubSubscriberTemplate.this`. Just curious , how come you  rather made a private class than passing the PubSubSubscriberTemplate in the constructor and having this be a separate class? If a whole PubSub template is too much functionality, perhaps you could just provide two `Consumer`s that do the ack and the modify ack deadline. It just felt a little strange to see an implementation of a message here in the template.

Reply:
When you move it into the top-level class, you get a class tangle which we would like to avoid.
There is nothing wrong having this class as an inner one as long as we have an `interface`  as a contract.

However I agree with the `Consumer` approach...

Reply:
Having `PulledAcknowledgeablePubsubMessage` as an inner class works out well here in my opinion. We're coding to an interface, and the whole `PulledAcknowledgeablePubsubMessage` is just an implementation detail completely invisible to the users.

Regarding the `Consumer` approach, I don't fully understand the vision with that on the pull side. Can you elaborate?

Reply:
Yes, of course: 

So it seems like this is a `private class` rather than a `private static class` because you would like to access the specific PubSubSubscriberTemplate instance that contains this inner class. However, you only need this PubSubSubscriberTemplate instance to do 2 very specific things, and you don't actually need the full functionality of it. So , the idea was to instead make this a private static class and pass those 2 specific operations you need in as two `Consumer`s. 

I'm just elaborating on the idea here, not saying it's blocking or anything. I'm fine with it as it is, too.